puppet/lib/dependabot/puppet/file_updater/requirement_replacer.rb

```diff
@@ -0,0 +1,211 @@
+          return updated_content if length_change.zero?
+
+          updated_lines = updated_content.lines
+          updated_line_index =
+            updated_lines.length.
+            times.find { |i| content.lines[i] != updated_content.lines[i] }
+          updated_line = updated_lines[updated_line_index]
+
+          updated_line =
+            if length_change.positive?
+              updated_line.sub(/(?<=\s)\s{#{length_change}}#/, "#")
+            elsif length_change.negative?
+              updated_line.sub(/(?<=\s{2})#/, " " * length_change.abs + "#")
+            end
+
+          updated_lines[updated_line_index] = updated_line
+          updated_lines.join
+        end
+
+        def length_change
+          unless previous_requirement.start_with?("=")
+            return updated_requirement.length - previous_requirement.length
+          end
+
+          updated_requirement.length -
+            previous_requirement.gsub(/^=/, "").strip.length
+        end
+
+        class Rewriter < Parser::TreeRewriter
+          # TODO: Ideally we wouldn't have to ignore all of these, but
+          # implementing each one will be tricky.
+          SKIPPED_TYPES = %i(send lvar dstr begin if splat const or).freeze
+
+          def initialize(dependency:, updated_requirement:,
+                         insert_if_bare:)
+            @dependency          = dependency
+            @updated_requirement = updated_requirement
+            @insert_if_bare      = insert_if_bare
+          end
+
+          def on_send(node)
+            return unless declares_targeted_module?(node)
+
+            req_nodes = node.children[3..-1]
+            req_nodes = req_nodes.reject { |child| child.type == :hash }
+
+            return if req_nodes.none? && !insert_if_bare?
+            return if req_nodes.any? { |n| SKIPPED_TYPES.include?(n.type) }
+
+            quote_characters = extract_quote_characters_from(req_nodes)
+            space_after_specifier = space_after_specifier?(req_nodes)
+            use_equality_operator = use_equality_operator?(req_nodes)
+
+            new_req = new_requirement_string(
+              quote_characters: quote_characters,
+              space_after_specifier: space_after_specifier,
+              use_equality_operator: use_equality_operator
+            )
+            if req_nodes.any?
+              replace(range_for(req_nodes), new_req)
+            else
+              insert_after(range_for(node.children[2..2]), ", #{new_req}")
+            end
+          end
+
+          private
+
+          attr_reader :dependency, :updated_requirement
+
+          def insert_if_bare?
+            @insert_if_bare
+          end
+
+          def declaration_methods
+            %i(mod)
+          end
+
+          def declares_targeted_module?(node)
+            return false unless declaration_methods.include?(node.children[1])
+
+            node.children[2].children.first == dependency.name.gsub('-', '/')
```

Comment:
Hi @greysteil I added a commit here that fixes this line to account for the difference in names between the syntax used by Puppetfiles and the naming requirements on the Puppet Forge. In the Puppetfile, there is a `/` splitting the author name and the module name, in the Puppet Forge there is a `-` as a separator. This gsub was in the original set of code, but was lost when we moved to the Ruby AST methods from the other parts of this repo.

While this is the only place that compares the module name, I'm not sure this is the best place to hold this logic. Should this be higher up the chain?